configuration-rendering-controller/README.md

```diff
@@ -0,0 +1,221 @@
+are all for the MC and they are all reconciled by the unique `app-operator`, which has RBAC rights to access the
+generated CMs and Secrets in any namespace.
+
+For workload cluster `app-operators` that might need access to configurations in the future, we could either give
+access to the namespace for CMs and Secrets that contains all rendered configs - if they would use the same -
+or render the workload cluster specific level to the given namespace. For one, the CR should be able to support that
+scope change. For another we do not have that implemented in the current system. The main issue here however is not
+a problem with this system, but with what we can share freely with customer workloads. For example, `shared-configs`
+is still kept private, because it contains - even if each encrypted with MC unique keys - minor Giant Swarm sensitive
+information. They are required by vintage to my knowledge, but they are there nevertheless and a security audit
+should be carried before we make that repository public.
+
+### Extending configuration management in the future
+
+Currently, we mostly utilise Catalog configuration for cluster apps. I think the reason for it is that it is not
+possible at the moment to utilise the configuration system when we deploy such workloads, so the configuration
+drifted into the catalogs, as the `cluster` catalog has well-defined scope for such apps.
+
+With easy access to configuration management, these catalog configs could simply drift into the config structure
+for the apps that need them on the MC specific level, and we could then simply get rid of them.
+
+## Implementation approaches
+
+### Framework
+
+The new controller should be written using [kubebuilder](https://book.kubebuilder.io/). It is on our roadmap to
+[Remove `operatorkit` as a dependency](https://github.com/giantswarm/roadmap/issues/3722) and this would be a great
+opportunity to get some hands-on experience with `kubebuilder`, as most of those operators that need and upgrade
+is owned by Team HoneyBadger.
+
+### Possibility to utilize Flux source-controller
+
+It is already implemented in `konfigure` to fetch the sources from Flux `source-controller` with `include`
+feature support [here](https://github.com/giantswarm/konfigure/pull/278).
+
+It would potentially be best to not depend on Flux either, but since we already have it implemented, and we would
+need to do something very similar to fetch and assemble the sources, we could cut this corner now and keep
+using Flux.
+
+There is potential to later implement custom DSL to describe how to merge configurations, which could make sense
+if we want to introduce many different layers in the future (catalogs, workload-cluster, app instance, etc. specific
+configs). For now, it makes sense to keep our current logic as is to reduce time to implement and friction.
+
+### Defining scope of configs to render
```

Comment:
If we mean here "for which App CRs to render the config", I think we should mdoel it after `selectors` in general, based on `nameSelector` or `labelSelector`. The default value is `- '*'` and can be overriden to something more specific, like:

```
target:
  - labelSelector:
      key: app.kubernetes.io/name
      value: flux.*
```

Reply:
It is not about App CRs. The config renderer does not know anything about App Platform. It is to define a sub scope of the config structure to render.

Reply:
Yeah, not App CRs, app names from the config repo, so this can't be based on labels. I wonder if we can do it better/more future-proof than just matching on the app config name. What I'm a bit afraid of is that in almost every usage scenario, the decision will be `.*` and the controller will generate a shitload (all) possible CM&S every single time and that can become a performance problem. Maybe it's worth passing the full "path" in the config repo of the most specific file merged in the config, so something like `my-repo/my-mc/(my-namespace)/my-app`? But this can be hard to work with.

Another problem: imagine we introduce, in the future, "namespace" scope of the config repo structure. 
- How do we select which namespace scoped configs are selected? From `targetNamespace`? From an extra selector (so an `and` between app name and namespace name)?
- How do we render the config? Do we just skip namespace scoped configs and render every config matching the app name pattern anyway or do we don't render it, as there's no namespace name match?

Again, I'm not saying this is wrong, but I'm a bit worried that rendering configs "blind", without any info about what's in the cluster, might cause some hard to solve problems in the future. The separation between config generation and usage is really nice, but it's not the key part of the solution here...